### PR TITLE
feat[FX-3212]: Add Algolia indexing to Articles

### PR DESCRIPTION
**Pull request by araujobarret**
[TICKET](https://artsyproduct.atlassian.net/browse/FX-3213)

- Add Algolia search module
- Index Articles in Algolia when the article is published
- Remove the index in Algolia when the article is not published

### CODE REVIEW

File: src/api/apps/articles/model/distribute.coffee [hunk 1]
```diff
@@ -96,6 +97,9 @@ moment = require 'moment'
       section.body
   tags = article.tags
   tags = tags.concat article.vertical.name if article.vertical
+  
+  #612fd7f526b2a7001fd62cb7
```
Comment (anandaroop) [inline src/api/apps/articles/model/distribute.coffee:101]: Oops 😄 

```suggestion
```

**Review by anandaroop — APPROVED** [review]
Nice work @araujobarret 🌟 

I only see some minor formatting issues, and one indexed attribute that I forgot to guide you towards. I'll go ahead and merge in the fixes for these via Github suggestions. 

And then I'm fine to merge this PR despite the CodeCov noise, which I think might be spurious in this case anyways.

File: scripts/reindex_articles.coffee [hunk 2]
```diff
@@ -20,11 +20,17 @@ db.articles.find({}).toArray (err, articles) ->
...
-  indexForSearch Article.present(article), () =>
-    console.log('indexed ' + article.id or article._id)
-    cb()
+  console.log('indexing', article._id)
+  articlePresent = Article.present(article) 
+  indexForSearch articlePresent, () =>
+    console.log('indexed on ElasticSearch', article.id or article._id)
+    if (articlePresent.published or articlePresent.scheduled_publish_at) and articlePresent.indexable
+      indexForAlgolia articlePresent, () =>
+        console.log('indexed on Algolia', article.id or article._id)
```
[the remark this answers is not in the document]
Comment (araujobarret) [reply scripts/reindex_articles.coffee]: Thanks for the review @anandaroop 🙌 
Just a heads up, when we use console.log with commas JS already puts an empty space after the string 😄

[the remark this answers is not in the document]
Comment (anandaroop) [reply scripts/reindex_articles.coffee]: Strange it seemed different for me locally, but no big deal in any case! 😄

**Comment by anandaroop** [comment]
I'll go ahead and merge this now and test things out on staging 🙌🏽

**Pull request merged**